File: src/main/java/architecture/goldenboughs_lib/init/tag/LibItemTags.java
package architecture.goldenboughs_lib.init.tag;

import architecture.goldenboughs_lib.core.Lib;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.tags.ItemTags;
import net.minecraft.tags.TagKey;
import net.minecraft.world.item.Item;

public final class LibItemTags {
	/**
	 * E.G.O
	 */
	public static final TagKey<Item> EGO = createTag("ego");

	private static TagKey<Item> createTag(String name) {
		return createTag(Lib.modRl(name));
	}

	private static TagKey<Item> createTag(ResourceLocation location) {
		return ItemTags.create(location);
	}

	private static TagKey<Item> createCTag(String name) {
		return createTag(ResourceLocation.fromNamespaceAndPath("c", name));
	}

	private static TagKey<Item> createMcTag(String name) {
		return createTag(ResourceLocation.withDefaultNamespace(name));
	}
}

File: src/main/kotlin/architecture/goldenboughs_lib/init/tag/LibItemTags.kt
package architecture.goldenboughs_lib.init.tag

import architecture.goldenboughs_lib.util.LibUtil.modRl
import architecture.goldenboughs_lib.util.LibUtil.rlOf
import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.ItemTags
import net.minecraft.tags.TagKey
import net.minecraft.world.item.Item

object LibItemTags {
	/**
	 * E.G.O
	 */
	@JvmField
	val EGO: TagKey<Item?> = createTag("ego")

	private fun createTag(name: String): TagKey<Item?> {
		return createTag(modRl(name))
	}

	private fun createTag(location: ResourceLocation): TagKey<Item?> {
		return ItemTags.create(location)
	}

	private fun createCTag(name: String): TagKey<Item?> {
		return createTag(rlOf("c", name))
	}

	private fun createMcTag(name: String): TagKey<Item?> {
		return createTag(ResourceLocation.withDefaultNamespace(name))
	}
}
